### PR TITLE
REP for large-scale shuffle on Ray

### PR DESCRIPTION
**Pull request by stephanie-wang**
This proposes enhancements to better support large-scale shuffle on Ray through the Datasets library.

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
LGTM. @scv119 take another pass?

File: reps/2022-03-09-shuffle.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+This proposal will not include any API changes other than a way to choose the shuffle implementation in Datasets.
+
+## Test Plan and Acceptance Criteria
+The proposal should discuss how the change will be tested **before** it can be merged or enabled. It should also include other acceptance criteria including documentation and examples. 
+
+We plan to benchmark Datasets using the following workloads:
+1. Shuffling data loader for ML ingest.
+2. Sorting.
+3. Groupby, using the [h2oai benchmark](https://h2oai.github.io/db-benchmark/).
+
```
Comment (scv119) [inline reps/2022-03-09-shuffle.md:95]: optional: should we include chaos test as failure is more common running in large scale?

**Review by scv119 — APPROVED** [review]
Looks good to me!

File: reps/2022-03-09-shuffle.md (hunk 1 above)
Comment (stephanie-wang) [reply reps/2022-03-09-shuffle.md:95]: Good idea, yes I'll add it.

**Pull request merged**